src/sqlfluff/core/rules/base.py

```diff
@@ -484,149 +586,107 @@ def _eval(self, context: RuleContext) -> EvalResultType:
 
     def crawl(
```

Comment:
This function is no longer recursive -- the recursion has moved to the `CrawlBehavior` helper class.